tensorflow/python/data/experimental/ops/get_single_element.py

```diff
@@ -20,31 +20,175 @@
+  CLASSES = [
+    "T-shirt/top", "Trouser", "Pullover", "Dress", "Coat", "Sandal",
+    "Shirt", "Sneaker", "Bag", "Ankle Boot"
+  ]
+
+  def make_dataset(images, labels):
+    ds = tf.data.Dataset.from_tensor_slices((images, labels))
+    ds = ds.shuffle(buffer_size=1024)
+    ds = ds.map(lambda x, y: (x / 255, y))
+    return ds.batch(batch_size=128)
+
+  def make_model():
+    # Add the layers
+    model = tf.keras.Sequential([
+      tf.keras.layers.Flatten(),
+      tf.keras.layers.Dense(10)
+    ])
+    # Compile the model
+    model.compile( optimizer='adam', metrics=['accuracy'],
+      loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    )
+    return model
+
+  model = make_model()
+  train_data = make_dataset(x_train, y_train)
+  test_data = make_dataset(x_test, y_test)
+  model.fit(train_data, validation_data=test_data, epochs=10)
+
+  # There is a need to pre-process the data before inference as the model
+  # will throw exceptions if the input is not of the desired shape. Thus, to
+  # serialize and save the pre-processing functionality along with the model,
+  # a `PreprocessingModel` is defined with a custom serving function called
+  # `serving_fn`.
+
+  class PreprocessingModel(tf.keras.Model):
+    def __init__(self, model):
+      super().__init__(self)
+      self.model = model
+
+    @tf.function(input_signature=[tf.TensorSpec([None, 28, 28], dtype=tf.uint8)])
+    def serving_fn(self, data):
+      # Strictly speaking, you do not need to use `tf.data` here at all, as simple
+      # `data / 255` would suffice. However, the point here is to illustrate that
+      # `tf.data` can be leveraged for optimized processing operations.
+      ds = tf.data.Dataset.from_tensor_slices(data)
+      ds = ds.map(lambda x: x / 255, num_parallel_calls=16)
+      ds = ds.batch(batch_size=16)
+      probabilities = self.model(tf.data.experimental.get_single_element(ds))
+      return tf.argmax(probabilities, axis=-1)
+
+  preprocessing_model = PreprocessingModel(model)
+  fmnist_save_path = "fmnist"
+
+  tf.saved_model.save(preprocessing_model, fmnist_save_path,
+                signatures={'serving_default': preprocessing_model.serving_fn})
+  ```
+
+  The saved model can now be served using `tensorflow-serving`.
+
+  ```console
+  $ apt-get install tensorflow-model-server
+  $ nohup tensorflow_model_server \
+    --rest_api_port=8501 \
+    --model_name=fashion_model \
+    --model_base_path="fmnist" >server.log 2>&1
+  ```
+  For additional details, refer: https://www.tensorflow.org/tfx/serving/serving_basic.
+
+  Estimators:
+  -----------
+
+  In the case of estimators, you need to generally define a `serving_input_fn`
+  which would require the features to be processed by the model while
+  inferencing.
+
+  For example:
+
+  ```python
+  def serving_input_fn():
+    # The function transforms the raw features of the data that is fed to
```

Comment:
remove comments that are not related to tf.data (such as this one) from this function ... the documentation of `get_single_element` should focus on tf.data specific logic and there is no need to explain tf.estimator specific logic

Reply:
yes, makes sense.